### PR TITLE
Fix various mistakes in devfile 2.0.x docs

### PR DESCRIPTION
**Pull request by sleshchenko**
This PR fixes various mistakes in devfile 2.0.x docs, each lives in their dedicated self-descriptive commit. Please merge with rebase to save the history.

### CODE REVIEW

**Review by johnmcollier — APPROVED** [review]
Changes look good to me.

**Review by amisevsk — APPROVED** [review]

File: docs/modules/user-guide/partials/proc_defining-endpoints.adoc [hunk 1]
```diff
@@ -63,9 +61,7 @@ components:
 
 Here, there are two container, each defining a single endpoint. Endpoint is an accessible port that can be made accessible inside the workspace or publicly (example, from the UI). Each endpoint has a name and port, which is the port on which certain server running inside the container is listening. The following are a few attributes that you can set on the endpoint:
 
-* `discoverable`: If an endpoint is discoverable, it means that it can be accessed using its name as the host name within the workspace containers (in the Kubernetes parlance, a Service is created for it with the provided name).
-
-* `public`: The endpoint will also be accessible outside of the workspace: such endpoint can be accessed from the {prod-short} user interface and is always exposed on port `80` or `443` (depending on whether `tls` is enabled in {prod-short}).
+* `exposure`: When its value is `public`, the endpoint is accessible outside the workspace. Users can access such an endpoint from the {prod-short} user interface and is always exposed on port `80` or `443` depending on whether TLS is enabled in {prod-short}.
```
Comment (jc-berger) [inline docs/modules/user-guide/partials/proc_defining-endpoints.adoc:64]: When I go to view the rendered file, I see that the text still reads, "Users can access such an endpoint from the {prod-short} user interface and is always exposed on port `80` or `443` depending on whether TLS is enabled in {prod-short}." 

Basically, `{prod-short}` doesn't render properly. I would simply write out devfiles. Also, I would rephrase the second part of the sentence, so the whole thing would read:

Users can access such an endpoint from the devfile user interface, and the endpoint is always exposed on port `80` or `443` depending on whether TLS is enabled in the devfile"

Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_defining-endpoints.adoc:64]: If `{prod-short}` isn't a concern (like maybe GH simply isn't rendering the output right), then still change the latter half of the sentence...the original phrasing might be read as "users are exposed on port..."

Comment (sleshchenko) [reply docs/modules/user-guide/partials/proc_defining-endpoints.adoc:64]: I think `{prod-short}` here is from Eclipse Che which may be also CodeReadyWorkspace. I'll rewrite in the suggested way without using prod-short

Comment (sleshchenko) [reply docs/modules/user-guide/partials/proc_defining-endpoints.adoc:64]: Well, there are a few more prod-short in other files. and antora-playbook has configured value `Devfile`, it should be fine.
I'll check if it's rendered in other places and create an issue if no

Comment (sleshchenko) [reply docs/modules/user-guide/partials/proc_defining-endpoints.adoc]: prod-short if rendered correctly in https://docs.devfile.io/devfile/2.1.0/user-guide/defining-kubernetes-resources.html, like `When running Devfile on a Kubernetes cluster` where Devfile is prod-short

Comment (jc-berger) [reply docs/modules/user-guide/partials/proc_defining-endpoints.adoc]: Okay, if it's rendered properly, it's not a concern :) Thanks for the follow up!
